### PR TITLE
fix: Reduce payload in event-type/getByViewer

### PR DESCRIPTION
**Pull request by hariombalhara**
Reduce payload size.

With a quick check I can see that the payload for `teampro` reduced from 1.4KB to 800bytes around 42% decrease in payload.

After uploading avatar for team and user and adding booking fields the payload on main was 3.5 KB and in this branch it is  1.3KB  which is more than 60% improvement

### CODE REVIEW

**Comment by hariombalhara** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #13580** <a href="https://app.graphite.dev/github/pr/calcom/cal.com/13580?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/calcom/cal.com/13580?utm_source=stack-comment).

**Review by emrysal — DISMISSED** [review]
👍

**Review by exception — DISMISSED** [review]

**Review by keithwillcode — DISMISSED** [review]

File: packages/lib/server/repository/eventType.ts [hunk 1]
```diff
@@ -33,6 +33,52 @@ const userSelect = Prisma.validator<Prisma.UserSelect>()({
   defaultScheduleId: true,
 });
 
+// We might need to relax it more in future as per requirements
+const eventTypeSelect = Prisma.validator<Prisma.EventTypeSelect>()({
```
Comment (hariombalhara) [inline packages/lib/server/repository/eventType.ts:37]: Select non JSON columns only here except those that are needed like metadata.

File: packages/features/eventtypes/components/EventTypeDescription.tsx [hunk 2]
```diff
@@ -16,11 +16,10 @@ import { Clock, Users, RefreshCw, Clipboard, Plus, User, Lock } from "@calcom/ui
 export type EventTypeDescriptionProps = {
   eventType: Pick<
     z.infer<typeof EventTypeModel>,
-    Exclude<keyof typeof baseEventTypeSelect, "recurringEvent"> | "metadata"
+    Exclude<keyof typeof baseEventTypeSelect, "recurringEvent"> | "metadata" | "seatsPerTimeSlot"
```
Comment (hariombalhara) [inline packages/features/eventtypes/components/EventTypeDescription.tsx:19]: Fixes ts error

File: packages/trpc/server/routers/viewer/eventTypes/getByViewer.handler.ts [hunk 3]
```diff
@@ -115,7 +115,7 @@ export const getByViewerHandler = async ({ ctx, input }: GetByViewerOptions) =>
           })
       )
     ),
-    metadata: eventType.metadata ? EventTypeMetaDataSchema.parse(eventType.metadata) : undefined,
+    metadata: eventType.metadata ? EventTypeMetaDataSchema.parse(eventType.metadata) : null,
```
Comment (hariombalhara) [inline packages/trpc/server/routers/viewer/eventTypes/getByViewer.handler.ts:118]: Fixes ts error.

**Review by exception — APPROVED** [review]

**Pull request merged**